src/citrine/builders/design_spaces.py

```diff
@@ -186,6 +199,13 @@ def cartesian_join_design_spaces(
     if len(all_keys) != len(set(all_keys)):
         raise ValueError('Duplicate keys are not allowed across design spaces')
 
+    # Check that the grid size is small enough to not cause memory issues
+    grid_size = np.prod([len(ds.data) * len(ds.data[0]) for ds in subspaces])
```

Comment:
Here, `grid_size` seems to be trying to be the number of cells (rows x columns).  However, I don't think its quite right.  I think you want np.prod([len(ds.data) for ds in subspaces]) * sum([len(ds.data[0]) for ds in subspaces]) since the total number of columns is just the sum (not the product) of the subspaces.

Reply:
Changed to (correctly) calculate total grid size.